### PR TITLE
refactor: make throttle controller decoupled from dsn_runtime

### PR DESCRIPTION
**Pull request by levy5307**
class rpc_message, which is a type in dsn_runtime, is used in throttle controller. So throttle controller is coupled with dsn_runtime now.
In this pull request, I want to decouple throttle controller from dsn_runtime,  so we can move throttle controller to utils in later pull requests.

### CODE REVIEW

**Comment by vagetablechicken** [comment]
I'm confused with the title 'decouple from dsn_runtime', maybe 'cleanup argument list' is better?

**Comment by levy5307** [comment]
> I'm confused with the title 'decouple from dsn_runtime', maybe 'cleanup argument list' is better?

I don't think so, because my main purpose is to do decouple work.

File: src/dist/replication/lib/throttling_controller.h [hunk 1]
```diff
@@ -83,7 +81,7 @@ class throttling_controller
     // do throttling control, return throttling type.
     // 'delay_ms' is set when the return type is not PASS.
     throttling_type
-    control(const message_ex *request, int32_t request_units, /*out*/ int64_t &delay_ms);
+    control(const int64_t client_timeout, int32_t request_units, /*out*/ int64_t &delay_ms);
```
Comment (hycdong) [inline src/dist/replication/lib/throttling_controller.h:84]: I suggest using `client_timeout_ms`

Comment (levy5307) [reply src/dist/replication/lib/throttling_controller.h:84]: good idea.

**Pull request merged**